connaisseur/util.py

```diff
@@ -1,5 +1,15 @@
+import os
+from connaisseur.exceptions import InvalidFormatException
+
+
 def normalize_delegation(delegation_role: str):
     prefix = "targets/"
     if not delegation_role.startswith(prefix):
         delegation_role = prefix + delegation_role
     return delegation_role
+
+
+def safe_path_exists(base_dir: str, path: str):
+    if os.path.commonprefix((os.path.realpath(path), base_dir)) != base_dir:
+        raise InvalidFormatException("potential path traversal.", {"path": path})
+    return os.path.exists(path)
```

Comment:
nice!

Just a random thought, a generalization of this would be:

```python
def safe_path_func(callback, base_dir: str, path: str, *args, **kwargs):
    if os.path.commonprefix((os.path.realpath(path), base_dir)) != base_dir:
        raise InvalidFormatException("potential path traversal.", {"path": path})
    return callback(path, *args, **kwargs)
```

... to be called with `safe_path_func(os.path.exists, ...)`. You could also replace `with open(...): by `with safe_path_func(open, ...):` etc.

Note that this is very much like a function decorator (or context manager), so there may be more elegant ways.

---

I also just realized that `InvalidFormatException(BaseConnaisseurException)` takes a `context` argument. It would be more pythonic to declare `BaseConnaisseurException.__init__(self, message: str, **kwargs)`, then set `self.context = kwargs`.

As a result, the exception here could be raised as `raise InvalidFormatException(..., path=path)`.

Reply:
The exception stuff would've to be done for all exceptions, so I think it's best to do it in a seperate refactoring PR including the deletion of the namespeace from above.